app/models/user.rb

```diff
@@ -9,7 +9,7 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable, :recoverable, :rememberable, :trackable, :validatable,
          authentication_keys: { username: false, email: true }
 
-  validates :username, uniqueness: true, allow_blank: true,
+  validates :username, uniqueness: true, allow_blank: false,
```

Comment:
We allow users to have blank usernames though, it gets saved as an empty string i believe. The usernames are currently only to make the url's look nicer, so we left them optional.